src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderResource.java

```diff
@@ -608,20 +607,10 @@ public Page<CaseNote> getOffenderCaseNotes(@PathVariable("offenderNo") @Paramete
                                                @RequestParam(value = "type", required = false) @Parameter(description = "Filter by case note type", example = "GEN") final String type,
                                                @RequestParam(value = "subType", required = false) @Parameter(description = "Filter by case note sub-type", example = "OBS") final String subType,
                                                @RequestParam(value = "prisonId", required = false) @Parameter(description = "Filter by the ID of the prison", example = "LEI") final String prisonId,
+                                               @RequestParam(value = "caseNoteTypeSubTypes", required = false) @Parameter(description = "Filter by list of case note types and optional case note sub types separated by plus", example = "KA+KE,OBS,POMK+GEN") final List<String> caseNoteTypeSubTypes,
                                                @ParameterObject @PageableDefault(sort = {"occurrenceDateTime"}, direction = Sort.Direction.DESC) final Pageable pageable) {
+        return caseNoteService.getCaseNotes(offenderNo,from,to,type,subType,prisonId,caseNoteTypeSubTypes,pageable);
```

Comment:
I think it was cleaner before - with the builder taking the parameters here and then passing through to the case notes service.  I'd let the builder take the `type`, `subType` and `typesSubTypes` and then use all of those to construct the `caseNoteFilter`. Any validation can then be introduced in the builder too.

Reply:
Refactored to use custom builder with validation.